packages/create-ponder/src/etherscan.ts

```diff
@@ -278,12 +270,36 @@ const getContractAbiAndName = async (
     address: contractAddress,
   });
   if (apiKey) searchParams.append("apikey", apiKey);
-  const data = await fetchEtherscan(`${apiUrl}?${searchParams.toString()}`);
 
-  const abi = data.result[0].ABI as string;
-  const contractName = data.result[0].ContractName as string;
+  const warnings: string[] = [];
+  let abi: Abi;
+  let contractName: string;
+
+  try {
+    const data = await fetchEtherscan(`${apiUrl}?${searchParams.toString()}`);
+
+    const rawAbi = data.result[0].ABI as string;
+    if (rawAbi === "Contract source code not verified") {
+      warnings.push(
+        `Contract ${contractAddress} is unverified or has an empty ABI.`,
+      );
+      abi = [];
+    } else {
+      abi = JSON.parse(rawAbi);
+    }
+
+    contractName = data.result[0].ContractName;
```

Comment:
What if this is undefined

Reply:
Added fallback. And this is in a block that gets caught so should be OK if it's something weird / an error.